src/Core/Module/ModuleManager.php

```diff
@@ -0,0 +1,355 @@
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Module;
+
+use Exception;
+use Module as LegacyModule;
+use PrestaShop\PrestaShop\Adapter\HookManager;
+use PrestaShop\PrestaShop\Adapter\Module\AdminModuleDataProvider;
+use PrestaShop\PrestaShop\Adapter\Module\ModuleDataProvider;
+use PrestaShop\PrestaShop\Core\Addon\Module\ModuleInterface as AddonModuleInterface;
+use PrestaShopBundle\Event\ModuleManagementEvent;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class ModuleManager implements ModuleManagerInterface
+{
+    /** @var ModuleRepository */
+    private $moduleRepository;
+
+    /** @var ModuleDataProvider */
+    private $moduleDataProvider;
+
+    /** @var AdminModuleDataProvider */
+    private $adminModuleDataProvider;
+
+    /** @var TranslatorInterface */
+    private $translator;
+
+    /** @var EventDispatcherInterface */
+    private $eventDispatcher;
+
+    /** @var HookManager */
+    private $hookManager;
+
+    /** @var Filesystem */
+    private $filesystem;
+
+    public function __construct(
+        ModuleRepository $moduleRepository,
+        ModuleDataProvider $moduleDataProvider,
+        AdminModuleDataProvider $adminModuleDataProvider,
+        TranslatorInterface $translator,
+        EventDispatcherInterface $eventDispatcher,
+        HookManager $hookManager
+    ) {
+        $this->filesystem = new Filesystem();
+        $this->moduleRepository = $moduleRepository;
+        $this->moduleDataProvider = $moduleDataProvider;
+        $this->adminModuleDataProvider = $adminModuleDataProvider;
+        $this->translator = $translator;
+        $this->eventDispatcher = $eventDispatcher;
+        $this->hookManager = $hookManager;
+    }
+
+    public function install(string $name): bool
+    {
+        if (!$this->adminModuleDataProvider->isAllowedAccess(__FUNCTION__)) {
+            throw new Exception($this->translator->trans(
+                'You are not allowed to install modules.',
+                [],
+                'Admin.Modules.Notification'
+            ));
+        }
+
+        $this->hookManager->exec('actionBeforeInstallModule', ['moduleName' => $name]);
+
+        $module = $this->moduleRepository->getModule($name);
+        $installed = $module->onInstall();
+        $this->dispatch(ModuleManagementEvent::INSTALL, $module);
```

Comment:
I think so because a module that override the method `install()` can call `parent::install()` before doing its own custom code. If the `parent::install()` went well but not the custom part, it will return false but in fact the module is considered installed for PrestaShop.